addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1474,6 +1474,13 @@ def channel_join(self):
         """
         self._add_members(users=self.env.user)
 
+    def _delete_sub_channel(self):
+        self.ensure_one()
+        body = Markup('<div data-oe-type="thread_deletion" class="o_mail_notification">%s</div>') % self.name
+        self.parent_channel_id.message_post(body=body, message_type="notification", subtype_xmlid="mail.mt_comment")
+        # sudo: discuss.channel - skipping ACL for users who created the thread
+        self.sudo().unlink()
```

Comment:
You’re checking the create_uid from the controller, but the sudo is applied inside the model. I think both the check and the sudo should be in the same method so that the justification for using sudo makes sense.

Maybe just use `channel.sudo()._delete_sub_channel()` from the controller?

Reply:
Agree, either comment should be along the lines of "_delete_sub_channel is a technical, access check should be done on the caller", or it should keep a similar comment as there is now but moved where the validation is done (also the comment is not complete, as many more things are checked rather than just who created the thread).

Reply:
Or actually `_delete_sub_channel` is useless as a method, it's only called once, just move the 3 lines into the controller.